pulsar-broker/src/main/java/org/apache/pulsar/broker/service/plugin/EntryFilterWithClassLoader.java

```diff
@@ -23,22 +23,27 @@
 import lombok.ToString;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.bookkeeper.mledger.Entry;
+import org.apache.pulsar.broker.ClassLoaderSwitcher;
 import org.apache.pulsar.common.nar.NarClassLoader;
 
 @Slf4j
 @ToString
 public class EntryFilterWithClassLoader implements EntryFilter {
     private final EntryFilter entryFilter;
     private final NarClassLoader classLoader;
+    private final boolean classLoaderOwned;
 
-    public EntryFilterWithClassLoader(EntryFilter entryFilter, NarClassLoader classLoader) {
+    public EntryFilterWithClassLoader(EntryFilter entryFilter, NarClassLoader classLoader, boolean classLoaderOwned) {
         this.entryFilter = entryFilter;
         this.classLoader = classLoader;
+        this.classLoaderOwned = classLoaderOwned;
     }
 
     @Override
     public FilterResult filterEntry(Entry entry, FilterContext context) {
-        return entryFilter.filterEntry(entry, context);
+        try (ClassLoaderSwitcher switcher = new ClassLoaderSwitcher(classLoader)) {
```

Comment:
maybe follow see also https://github.com/apache/pulsar/pull/22790 's pattern is better

Reply:
I think that it is better to have a consistent code style and apply it all over the places.
Currently we are using this pattern.
We can remove that class in a later PR in all of the places.

This patch must be merged to all the active branches

Reply:
OK, I will remove the class and apply #22790 's pattern first.

Reply:
@eolivelli PTAL https://github.com/apache/pulsar/pull/22796

Reply:
done